events/viewsv2.py

```diff
@@ -2312,3 +2312,41 @@ def form_valid(self, form, **kwargs):
       print "saved"
       messages.success(self.request, "Thank you for completing this feedback form. We appreciate your input and valuable suggestions.")
       return HttpResponseRedirect(self.success_url)
+      
+  
+def verify_email(request):
+  template_name = 'verify_email.html'
+  context = {}
+  ci = RequestContext(request)
+  if request.method == 'POST':
+    email = request.POST.get('email').strip()
+    try:
+      user = User.objects.get(email = email)
+    except ObjectDoesNotExist:
+      context["notregistered"] = 1
+      return render_to_response(template_name, context, ci)
+    try:
+      student = Student.objects.get(user_id = user.id)
+      if student and student.verified:
+        messages.success(request, 'User is already verified')
+        return HttpResponseRedirect('/home')
+      if student and not student.verified and validate_email(student.user.email):
+        print "validating"
+        student.verified = True
+        student.user.is_active = True
+        student.user.save()
+        student.error = False
+        student.save()
+        messages.success(request, 'Student Email Id verified and account is activated.')
```

Comment:
No you don't make the active user active here. You are just resending sending the email, if not already verified. You can get ride of/modify this code.